h2o-py/tests/testdir_sklearn/pyunit_sklearn_regression_all_estimators.py

```diff
@@ -0,0 +1,170 @@
+from __future__ import print_function
+from collections import defaultdict
+from functools import partial
+from itertools import chain
+import importlib, inspect, os, sys
+
+import numpy as np
+from sklearn.datasets import make_regression
+from sklearn.metrics import r2_score
+from sklearn.model_selection import train_test_split
+
+import h2o
+from h2o.sklearn.wrapper import H2OConnectionMonitorMixin
+
+
+sys.path.insert(1, os.path.join("..",".."))
+from tests import pyunit_utils, Namespace as ns
+
+
+"""
+This test suite creates a default sklearn regression estimator for each H2O estimator.
+Then, it feeds them with H2O frames (more efficient and ensures compatibility with old API.)
+or with numpy arrays to provide the simplest approach for users wanting to use H2O like any sklearn estimator.
+"""
+
+seed = 2019
+init_connection_args = dict(strict_version_check=False, show_progress=True)
+
+scores = defaultdict(dict)
+
+
+def _ensure_connection_state(connected=True):
+    if connected:
+        # if we need a connection beforehand, create it if needed
+        H2OConnectionMonitorMixin.init_connection(init_connection_args)
+    else:
+        # if we want to start afresh, close everything first
+        H2OConnectionMonitorMixin.close_connection(force=True)
+
+
+
+def _get_data(format='numpy'):
+    X, y = make_regression(n_samples=100, n_features=10, n_informative=5, random_state=seed)
+    X_train, X_test, y_train, y_test = train_test_split(X, y, random_state=seed)
+    data = ns(X_train=X_train, X_test=X_test, y_train=y_train, y_test=y_test)
+    if format == 'h2o':
+        for k, v in data.__dict__.items():
+            setattr(data, k, h2o.H2OFrame(v))
+    return data
+
+
+def _get_default_args(estimator_cls):
+    defaults = dict(
+        H2OCoxProportionalHazardsRegressor=dict(),
+        H2OGeneralizedLinearRegressor=dict(family='gaussian', seed=seed),
+    )
+    return defaults.get(estimator_cls.__name__, dict(seed=seed))
+
+
+def _get_custom_behaviour(estimator_cls):
+    custom = dict(
+        H2ODeepLearningRegressor=dict(scores_may_differ=True),
```

Comment:
deep learning with default params (but with a seed) doesn't look reproducible: doesn't look normal

Reply:
deep learning is not reproducible by default even with a given seed

Reply:
always thought that is was on single node.
@ledell FYI

Reply:
yeah, not even on a single node - to make it reproducible you need to use flag `reproducible=true`, http://docs.h2o.ai/h2o/latest-stable/h2o-docs/data-science/deep-learning.html